src/pin.rs

```diff
@@ -0,0 +1,219 @@
+//! Mock digital InputPin and OutputPin implementations
+//! 
+//! ```
+//! use std::io::ErrorKind;
+//! 
+//! use embedded_hal_mock::MockError;
+//! use embedded_hal_mock::pin::{Transaction as PinTransaction, Mock as MockPin};
+//! use embedded_hal::digital::v2::{InputPin, OutputPin};
+//! 
+//! let err = MockError::Io(ErrorKind::NotConnected);
+//! 
+//! // Configure expectations
+//! let expectations = [
+//!     PinTransaction::get(true),
+//!     PinTransaction::get(true),
```

Comment:
I wonder if it would be better to have explicit pin states: `enum PinState { High, Low }`.

It would look a bit less confusing when creating the list of expectations:

```rust
 let expectations = [
    PinTransaction::get(PinState::High),
    PinTransaction::get(PinState::High),
];

assert_eq!(pin.is_high().unwrap(), true);
assert_eq!(pin.is_low().unwrap(), false);
```